Src/coffeescript/cql-execution/src/datatypes/interval.coffee

```diff
@@ -1,67 +1,212 @@
+    closed = @toClosed()
+    ThreeValuedLogic.and(
+      cmp.lessThanOrEquals(closed.low, item),
+      cmp.greaterThanOrEquals(closed.high, item)
+    )
 
-  includedIn: (item) ->
-    if item instanceof Interval
-      item.includes @
-    else
-      [uLow, uHigh] = @_getEndpointsAsUncertainties()
-      uItem = if item.toUncertainty? then item.toUncertainty() else new Uncertainty(item)
-      ThreeValuedLogic.and @lowClosed, @highClosed, uLow.equals(uHigh), uLow.equals(uItem), uHigh.equals(uItem)
+  properlyIncludes: (other) ->
+    if not (other instanceof Interval) then throw new Error("Argument to properlyIncludes must be an interval")
+    a = @toClosed()
+    b = other.toClosed()
+    ThreeValuedLogic.and(
+      cmp.lessThan(a.low, b.low),
+      cmp.greaterThan(a.high, b.high)
+    )
+
+  includes: (other) ->
+    if not (other instanceof Interval) then throw new Error("Argument to includes must be an interval")
+    a = @toClosed()
+    b = other.toClosed()
+    ThreeValuedLogic.and(
+      cmp.lessThanOrEquals(a.low, b.low),
+      cmp.greaterThanOrEquals(a.high, b.high)
+    )
+
+  includedIn: (other) ->
+    if not (other instanceof Interval) then throw new Error("Argument to includedIn must be an interval")
+    other.includes @
 
   overlaps: (item) ->
-    if item instanceof Interval
-      [uLow, uHigh] = @_getEndpointsAsUncertainties()
-      [uItmLow, uItmHigh] = item._getEndpointsAsUncertainties()
-      uLow.lessThanOrEquals(uItmHigh) and uHigh.greaterThanOrEquals(uItmLow)
+    closed = @toClosed()
+    [low, high] = if item instanceof Interval
+      itemClosed = item.toClosed()
+      [itemClosed.low, itemClosed.high]
     else
-      @includes item
+      [item, item]
+    ThreeValuedLogic.and(
+      cmp.lessThanOrEquals(closed.low, high),
+      cmp.greaterThanOrEquals(closed.high, low)
+    )
 
   overlapsAfter: (item) ->
+    closed = @toClosed()
+    high = if item instanceof Interval then item.toClosed().high else item
     ThreeValuedLogic.and(
-      @overlaps(item),
-      @toClosed().high.after(if item instanceof Interval then item.toClosed().high else item)
+      cmp.lessThanOrEquals(closed.low, high),
+      cmp.greaterThan(closed.high, high)
     )
 
   overlapsBefore: (item) ->
+    closed = @toClosed()
+    low = if item instanceof Interval then item.toClosed().low else item
     ThreeValuedLogic.and(
-      @overlaps(item),
-      @toClosed().low.before(if item instanceof Interval then item.toClosed().low else item)
+      cmp.lessThan(closed.low, low),
+      cmp.greaterThanOrEquals(closed.high, low)
     )
 
-  equals: (item) ->
-    if item instanceof Interval
-      [a, b] = [@toClosed(), item.toClosed()]
-      ThreeValuedLogic.and equals(a.low, b.low), equals(a.high, b.high)
+  areDateTimes = (x, y) ->
+    [x, y].every (z) -> z instanceof DateTime
+
+  areNumeric = (x, y) ->
+    [x, y].every (z) -> typeof z is 'number' or (z instanceof Uncertainty and typeof z.low is 'number')
```

Comment:
I'm just being lazy and not looking but why does't areDateTimes look to see if z is an uncertainty with a low with type DateTime the way that areNumeric does. Does Uncertainty not support DateTime